Document/0x05f-Testing-Authentication.md

```diff
@@ -155,10 +155,33 @@ Many Apps do not automatically logout a user, because of customer convenience. T
 
 #### Overview
 
+Password policy is a special countermeasure to avoid any application user on creating passwords that are easily guessable. 
 -- TODO [Provide a general description of the issue "Testing the Password Policy".] --
 
 #### Static Analysis
 
+When testing a password policy, a well-defined rule set can be used to verify that source code not contain any previously identified technical or logical security flew. A exemplary rule set (source: VT Password<sup>[1]</sup>) is given below:
+* AllowedCharacterRule - Does a password contain only a specific list of characters
+* AlphabeticalSequenceRule - Does a password contain an alphabetical sequence
+* CharacterCharacteristicRule - Does a password contain the desired mix of character types
+* DictionaryRule - Does a password match a word in a dictionary
+* DictionarySubstringRule - Does a password contain a word in a dictionary
+* DigitCharacterRule - Does a password contain a digit
+* HistoryRule - Does a password match a previous password, supports hashes
+* IllegalCharacterRule - Does a password contain an illegal character
+* LengthRule - Is a password of a certain length
+* LowercaseCharacterRule - Does a password contain a lowercase character
+* NonAlphanumericCharacterRule - Does a password contain a non-alphanumeric character
+* NumericalSequenceRule - Does a password contain a numerical sequence
+* RegexRule - Does a password match a regular expression
+* RepeatCharacterRegexRule - Does a password contain a repeated character
+* SequenceRule - Does a password contain a keyboard sequence
+* SourceRule - Does a password match the password from another system or source
+* QwertySequenceRule - Does a password contain a QWERTY keyboard sequence
+* UppercaseCharacterRule - Does a password contain an uppercase character
+* UsernameRule - Does a password contain a username
+* WhitespaceRule - Does a password contain whitespace
```

Comment:
I would remove this listing, as I think a password policy should allow any character as stated here, this should also be used as a reference:
https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Implement_Proper_Password_Strength_Controls 

Static Analysis of the password policy would involve server code and you could list down an example for a regex that might be used to check for a strong password. You could describe how you can check if the regex is correct through something like http://regexr.com/ and see what passwords are matching the regex